_posts/2024-08-14-adding-to-bioconda-quickguide.md

```diff
@@ -0,0 +1,468 @@
+However, the steps here should act as a good starting point.
+Furthermore, if you are planning to add someone else's tool or package to Bioconda, it's always good etiquette to ask or just inform the original authors that it will happen.
+
+_I would like to thank George Bouras (@gbouras13) for prompting the formalisation of my rough notes that he shared on Twitter where it became more popular than I expected._
+
+## TL;DR
+
+### Overview
+
+![Overview diagram of the steps of the steps described in this post.]({% link assets/images/2024-08-14-bioconda-guide/bioconda-guide-workflowdiagram_short.svg %})
+
+### Relevant Commands
+
+```bash
+## Create environment with conda building tools
+conda create -n bioconda-build -c conda-forge -c bioconda conda-build bioconda-utils greyskull
+conda activate bioconda-build
+
+## Clone repo and make branch
+git clone <your-forked-bioconda-recipes-repo-address>
+git switch -c add-<toolname>
+
+## Make recipe meta.yaml
+   ## Option 1: If using Greyskull
+   cd recipes/
+   greyskull <pypi/cran> > <toolname>/meta.yaml
+
+   ## Option 2: If not using Greyskull
+   mkdir recipes/<toolname>
+   touch recipes/<toolname>/meta.yaml
+
+## Lint recipe meta.yaml
+bioconda-utils lint recipes/ --packages pathphynder
+
+## Perform a local build (two options)
+   ## Option 1:
+   conda build recipes/<toolname>
+
+   ## Option 2:
+   bioconda-utils build --docker --mulled-test --packages <toolname>
+
+## Debugging
+   ## Option 1: conda-build
+   cd /<path>/<to>/<conda-install>/envs/<toolname>/conda-bld/linux-64
+   conda create -n <debugging-env-name> -c ./ <tool_name_as_in_recipe>
+   conda build recipes/<toolname> --keep-old-work
+
+   ## Option 2: bioconda-utils
+   docker run -t --net host --rm -v /tmp/tmp<randomletters>/build_script.bash:/opt/build_script.bash -v /<path>/<to>/<conda-install>/envs/<toolname>/conda-bld/:/opt/host-conda-bld -v /<path>/<to>/<recipes_local_clone>/recipes/<toolname>:/opt/recipe -e LC_ADDRESS=en_GB.UTF-8 -e LC_NAME=en_GB.UTF-8 -e LC_MONETARY=en_GB.UTF-8 -e LC_PAPER=en_GB.UTF-8 -e LANG=en_GB.UTF-8 -e LC_IDENTIFICATION=en_GB.UTF-8 -e LC_TELEPHONE=en_GB.UTF-8 -e LC_MEASUREMENT=en_GB.UTF-8 -e LC_TIME=en_GB.UTF-8 -e LC_NUMERIC=en_GB.UTF-8 -e HOST_USER_ID=1000 quay.io/bioconda/bioconda-utils-build-env-cos7:2.11.1 bash
+
+   conda mambabuild -c file:///opt/host-conda-bld --override-channels --no-anaconda-upload -c conda-forge -c bioconda -c defaults -e /opt/host-conda-bld/conda_build_config_0_-e_conda_build_config.yaml -e /opt/host-conda-bld/conda_build_config_1_-e_bioconda_utils-conda_build_config.yaml /opt/recipe/meta.yaml 2>&1
+   conda activate /opt/conda/conda-bld/<toolname_hash>/_build_env
+
+## Testing the Docker image artifact
+docker run -it <image_id_from_docker_images_command>
+```
+
+## Prerequisites
+
+1. Make a fork of the [bioconda-recipes](https://github.com/bioconda/bioconda-recipes/) GitHub repository, and clone this to our local machine.
```

Comment:
Will this still work if you shallow clone? git clone --depth 1 is 10x smaller (37.89 MiB vs 377.71 MiB)

Reply:
Yes, you can do - I don't see why it wouldn't work.

I specifically didn't go into too much detail into describing git because I sort of give it as an implicit requirement for doing this (the tool/package in most cases should be on git anyway...)

But I've added a footnote mentioning this :)